src/iptvsimple/Settings.cpp

```diff
@@ -23,192 +22,153 @@ void Settings::ReadFromAddon(const std::string& userPath, const std::string clie
-    m_udpxyPort = DEFAULT_UDPXY_MULTICAST_RELAY_PORT;
-  if (!XBMC->GetSetting("useFFmpegReconnect", &m_useFFmpegReconnect))
-    m_useFFmpegReconnect = false;
-  if (!XBMC->GetSetting("useInputstreamAdaptiveforHls", &m_useInputstreamAdaptiveforHls))
-    m_useInputstreamAdaptiveforHls = false;
-  if (XBMC->GetSetting("userAgent", &buffer))
-    m_userAgent = buffer;
+  m_transformMulticastStreamUrls = kodi::GetSettingBoolean("transformMulticastStreamUrls");
+  m_udpxyHost = kodi::GetSettingString("udpxyHost");
+  m_udpxyPort = kodi::GetSettingInt("udpxyPort");
+  m_useFFmpegReconnect = kodi::GetSettingBoolean("useFFmpegReconnect");
+  m_useInputstreamAdaptiveforHls = kodi::GetSettingBoolean("useInputstreamAdaptiveforHls");
+  m_userAgent = kodi::GetSettingString("userAgent");
 }
 
-ADDON_STATUS Settings::SetValue(const std::string& settingName, const void* settingValue)
+ADDON_STATUS Settings::SetValue(const std::string& settingName, const kodi::CSettingValue& settingValue)
 {
   // reset cache and restart addon
 
   std::string strFile = FileUtils::GetUserDataAddonFilePath(M3U_CACHE_FILENAME);
-  if (FileUtils::FileExists(strFile.c_str()))
+  if (FileUtils::FileExists(strFile))
     FileUtils::DeleteFile(strFile);
 
   strFile = FileUtils::GetUserDataAddonFilePath(XMLTV_CACHE_FILENAME);
-  if (FileUtils::FileExists(strFile.c_str()))
+  if (FileUtils::FileExists(strFile))
     FileUtils::DeleteFile(strFile);
 
   // M3U
   if (settingName == "m3uPathType")
-    return SetSetting<PathType, ADDON_STATUS>(settingName, settingValue, m_m3uPathType, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uPath")
-    return SetStringSetting<ADDON_STATUS>(settingName, settingValue, m_m3uPath, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uUrl")
-    return SetStringSetting<ADDON_STATUS>(settingName, settingValue, m_m3uUrl, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uCache")
-    return SetSetting<bool, ADDON_STATUS>(settingName, settingValue, m_cacheM3U, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "startNum")
-    return SetSetting<int, ADDON_STATUS>(settingName, settingValue, m_startChannelNumber, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "numberByOrder")
-    return SetSetting<bool, ADDON_STATUS>(settingName, settingValue, m_numberChannelsByM3uOrderOnly, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uRefreshMode")
-    return SetSetting<RefreshMode, ADDON_STATUS>(settingName, settingValue, m_m3uRefreshMode, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uRefreshIntervalMins")
-    return SetSetting<int, ADDON_STATUS>(settingName, settingValue, m_m3uRefreshIntervalMins, ADDON_STATUS_OK, ADDON_STATUS_OK);
-  if (settingName == "m3uRefreshHour")
-    return SetSetting<int, ADDON_STATUS>(settingName, settingValue, m_m3uRefreshHour, ADDON_STATUS_OK, ADDON_STATUS_OK);
+    m_m3uPathType = static_cast<PathType>(settingValue.GetInt());
```

Comment:
This is reading the value and not setting it.

Reply:
Not understand what you mean?